### PR TITLE
TrappedChest's MaterialData should be org.bukkit.material.Chest

### PR DESCRIPTION
**Pull request by Tonodus**
It's confusing that TrappedChests don't have `org.material.Chest` as `MaterialData`.
That is IMO a logical mistake and TrappedChests' facing direction can't be changed (by plugins).

This PR changes this without breaking any plugins/sever implementations/etc.

---

_Edited by turt2live_

**Related links:**
- Glowstone: GlowstoneMC/Glowstone#358
  - Glowstone PR is not required for operation/implementation

### CODE REVIEW

**Comment by turt2live** [comment]
Closing as this is no longer linked to an applicable Glowstone PR. Please respond back if this is still required (it doesn't look like it).

**Comment by Tonodus** [comment]
@turt2live 
Since https://github.com/GlowstoneMC/Glowstone/pull/358 has come alive, this should be re-opened. When this PR is pulled, the double chests implemented in the above PR should automatically connect right (only trap&trap or not trapped&not trapped).

File: src/main/java/org/bukkit/material/Chest.java [hunk 1]
```diff
@@ -51,6 +60,11 @@ public Chest(final int type, final byte data) {
     @Deprecated
     public Chest(final Material type, final byte data) {
         super(type, data);
+        isTrapped = type == Material.TRAPPED_CHEST;
+    }
+
+    public boolean isTrappedChest() {
+        return isTrapped;
```
Comment (turt2live) [inline src/main/java/org/bukkit/material/Chest.java]: Pretty sure you can check this dynamically.

**Comment by Tonodus** [comment]
@turt2live made it dynamically.

File: src/main/java/org/bukkit/material/Chest.java [hunk 2]
```diff
@@ -23,7 +27,6 @@ public Chest(BlockFace direction) {
     }
 
     /**
-     *
```
Comment (dequis) [inline src/main/java/org/bukkit/material/Chest.java:26]: Re-add this line to make the diff even smaller :3

**Comment by turt2live** [comment]
I don't know why I closed this in the first place... it's Glowstone agnostic...

File: src/main/java/org/bukkit/material/Chest.java [hunk 3]
```diff
@@ -53,6 +57,10 @@ public Chest(final Material type, final byte data) {
         super(type, data);
     }
 
+    public boolean isTrappedChest() {
```
Comment (turt2live) [inline src/main/java/org/bukkit/material/Chest.java:65]: Missing documentation

File: src/main/java/org/bukkit/material/Chest.java [hunk 4]
```diff
@@ -4,12 +4,16 @@
 import org.bukkit.block.BlockFace;
 
 /**
- * Represents a chest
+ * Represents a (trapped) chest
```
Comment (turt2live) [inline src/main/java/org/bukkit/material/Chest.java]: I would prefer this to be `Represents a chest or trapped chest` to avoid any potential confusion

**Comment by Tonodus** [comment]
@turt2live applied your suggestions.

**Comment by SpaceManiac** [comment]
Merged in 4e9cc32cffc2d26c69ebe10e3ff728f07a77aef5.

**Pull request closed without merging**